mirrord/agent/src/file.rs

```diff
@@ -516,4 +531,65 @@ impl FileManager {
         })
         .map_err(ResponseError::from)
     }
+
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub(crate) fn open_dir(&mut self, fd: u64) -> RemoteResult<OpenDirResponse> {
+        trace!("FileManager::open_dir -> fd {:#?}", fd,);
+        let path = match self
+            .open_files
+            .get(&fd)
+            .ok_or(ResponseError::NotFound(fd))?
+        {
+            RemoteFile::Directory(path) => path,
+            _ => panic!("Opened path is a file."),
```

Comment:
Panic is bad, this is a "valid" case where we return an error that translates to the appropriate libc error (`ENOTDIR`)